ygen/ir.go

```diff
@@ -158,11 +197,30 @@ type NodeDetails struct {
 	// the output code, using the output of the language-specific
 	// type mapping provided by calling the LangMapper interface.
 	LangType *MappedType
-	// MapPaths describes the paths that the output node should
+	// MappedPaths describes the paths that the output node should
 	// be mapped to in the output code - these annotations can be
 	// used to annotation the output code with the field(s) that it
 	// corresponds to in the YANG schema.
-	MapPaths [][]string
+	MappedPaths [][]string
+	// MappedModules describes the path elements' belonging modules that
+	// the output node should be mapped to in the output code - these
+	// annotations can be used to annotation the output code with the
+	// field(s) that it corresponds to in the YANG schema.
+	MappedModules [][]string
```

Comment:
Ah -- sorry, this is clearer as to why "mapped" is in the name. Maybe we can call this `MappedPathElementModules` - although that is quite wordy.... I'm OK with `MappedModules`.

Reply:
Ok I changed it to `MappedPathModules`, maybe it's a good compromise.

Reply:
SGTM.